src/components/pages/events/containers/NewEventPageContainer.js

```diff
@@ -17,10 +17,14 @@ export default compose(
   connect(
     state => ({
       neighborhoods: state.neighborhood.neighborhoods,
-      selectedAddress: state.geo.selectedAddress,
+      selectedAddress: state.geo.geoData && state.geo.geoData.closest_address,
```

Comment:
You could use lodash's `get` instead. then you don't have to handle undefined errors.